src/core/server/http/router/router.ts

```diff
@@ -344,12 +350,12 @@ type RequestHandlerEnhanced<
  * @public
  */
 export type RequestHandler<
-  P extends ObjectType,
-  Q extends ObjectType,
-  B extends ObjectType | Type<Buffer> | Type<Stream>,
+  P extends ObjectType | ValidateFunction<unknown>,
```

Comment:
would be good to add at least one usage example. in integration tests, e.g.
```typescript
interface Foo {
  foo?: number;
}
test('invalid params', async () => {
  const router = new Router('/foo', logger, enhanceWithContext);

  const validate = {
    params: (params: Foo) => {
       // TODO add an example with an error
      const value = typeof params.foo === 'number' ?  params.foo || 0;
      return { value };
    },
  };
  const handler: RequestHandler<typeof validate.params> = (context, req, res) => {
    console.log(req.body);
    return res.ok({ body: String(req.params.test) });
  };

  router.get(
    {
      path: '/path-1',
      validate,
    },
    handler
  );

  const validateKbnSchema = {
    params: schema.object({
      test: schema.string(),
    }),
  };
  const handlerKbnSchema: RequestHandler<typeof validateKbnSchema.params> = (context, req, res) => {
    console.log(req.body);
    return res.ok({ body: String(req.params.test) });
  };

  router.get(
    {
      path: '/path-1',
      validateKbnSchema,
    },
    handlerKbnSchema
  );
```

Reply:
do you mean this? https://github.com/elastic/kibana/pull/51919/files#diff-51730cbab06a430a940f78fc9e710bd7R290

Reply:
ah... I overlooked it and wrote my test instead. stupid me 🤣